src/test/java/com/aws/greengrass/authorization/AuthorizationHandlerTest.java

```diff
@@ -386,9 +413,33 @@ void GIVEN_AuthZ_handler_WHEN_service_registered_THEN_auth_lookup_for_star_resou
                 Permission.builder().principal("compA").operation("OpA").resource(null).build()));
     }
 
+    @Test
+    void GIVEN_AuthZ_handler_WHEN_service_registered_THEN_auth_lookup_for_wildcard_resource_works() throws Exception {
+        AuthorizationHandler authorizationHandler = new AuthorizationHandler(mockKernel, authModule, policyParser,
+                deviceConfiguration);
+        when(mockKernel.findServiceTopic(anyString())).thenReturn(mockTopics);
+        Set<String> serviceOps = new HashSet<>(Arrays.asList("OpA"));
+        authorizationHandler.registerComponent("ServiceA", serviceOps);
+
+        AuthorizationPolicy policy = getWildcardResourceAuthZPolicy();
+        authorizationHandler.loadAuthorizationPolicies("ServiceA", Collections.singletonList(policy),
+                false);
+        assertTrue(authorizationHandler.isAuthorized("ServiceA",
+                Permission.builder().principal("compA").operation("OpA").resource("abc123/def/asxyzds4/2/4/ghj").build()));
+
+        // Multiple levels don't work in '+'
+        assertThrows(AuthorizationException.class, () -> authorizationHandler.isAuthorized("ServiceA",
```

Comment:
shouldn't this only apply to MQTT?

If I put `+`/`#` in a resource which isn't MQTT, then I don't expect it to work as a wildcard. I think you need a way to signal whether to use MQTT-style lookups or not.

Reply:
So what would be the usage of '+' or '#' in Non MQTT-style lookup? you mean then we should treat them as normal characters?

Reply:
Right, if this is a secret resource for example, then + and # aren't wildcards, they're just characters.

Reply:
Ok, so right now. In case + is not between 2 '/'s (eg: /+/) and if # is not preceded by a '/' (eg: /#) they will be treated normally

Reply:
I don't think those are good rules. + and # should be treated as the wildcards that they are, regardless of `/`.

Reply:
The resource `#` should work in MQTT-style to be a catch-all, right?

Reply:
I think thats what happens in our IoT MQTT styling...not sure though

Reply:
No, I think it must always be on the last level

Reply:
Literally `#` should work as `*`, this is what I'm telling you. We don't care what cloud says; we are following closer to the MQTT spec than IoT Core.

Reply:
Oh okay, I see I didn't realize IoT Core was following different than MQTT spec.